File: CS/ReportGalleryWinForms/Form1.cs
using XpoReportGalleryStorage;
using DevExpress.XtraReports.Extensions;
using System.Windows.Forms;

namespace ReportGalleryWinForms {
    public partial class Form1 : Form {
        XpoReportGalleryStorageBase xpoGalleryStorage;
        public Form1() {
            InitializeComponent();
            xpoGalleryStorage = new XpoReportGalleryStorageSimple();
            //Uncomment thisline to register a gallery storage that saves and loads the changeditems only.
            //xpoGalleryStorage = new XpoReportGalleryStorageAdvanced();
            ReportGalleryExtension.RegisterExtensionGlobal(xpoGalleryStorage);
            FormClosed += OnFormClosed;
        }

        void OnFormClosed(object sender, FormClosedEventArgs e) {
            xpoGalleryStorage.Dispose();
        }
    }
}

File: VB/reporting-provide-custom-report-gallery-storage-in-report-designer/ReportGalleryWinForms/Form1.vb
Imports XpoReportGalleryStorage
Imports DevExpress.XtraReports.Extensions
Imports System.Windows.Forms

Namespace ReportGalleryWinForms

    Public Partial Class Form1
        Inherits Form

        Private xpoGalleryStorage As XpoReportGalleryStorageBase

        Public Sub New()
            InitializeComponent()
            xpoGalleryStorage = New XpoReportGalleryStorageSimple()
            'Uncomment thisline to register a gallery storage that saves and loads the changeditems only.
            'xpoGalleryStorage = new XpoReportGalleryStorageAdvanced();
            ReportGalleryExtension.RegisterExtensionGlobal(xpoGalleryStorage)
            AddHandler FormClosed, AddressOf OnFormClosed
        End Sub

        Private Overloads Sub OnFormClosed(ByVal sender As Object, ByVal e As FormClosedEventArgs)
            xpoGalleryStorage.Dispose()
        End Sub
    End Class
End Namespace
